### PR TITLE
 Use LGTM workflow in the config

### PR DESCRIPTION
**Pull request by Daverlo**
@aibaars

### CODE REVIEW

File: src/main/resources/js/lgtm-addon.js [hunk 1]
```diff
@@ -304,6 +210,43 @@ function updateConfig() {
...
+			AJS.messages.error("#message-context", {
+				title : 'The project ' + AJS.$(projectFieldId).select2('data').text
+						+ ' does not exist.',
+				closeable : true,
+				fadeout : true
+			});
+			return;
+		} else if (jqXHR.getResponseHeader("Error") === "issueType") {
+			AJS.messages.error("#message-context", {
+				title : 'The LGTM alert issue type doe not exist.',
```
Comment (Daverlo) [inline src/main/resources/js/lgtm-addon.js]: The error messages are temporal, could you help me find some better messages?

File: src/main/java/com/semmle/jira/addon/util/JiraUtils.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.ofbiz.core.entity.GenericEntityException;
+import org.ofbiz.core.entity.GenericValue;
+
+public class JiraUtils {
+
+  public static void addIssueTypeToProject(String projectKey, String issueTypeName)
```
Comment (aibaars) [inline src/main/java/com/semmle/jira/addon/util/JiraUtils.java]: I think this method should take a project ID as argument. That way you can move the `getProjectByCurrentKey` call out of this method. Same for issue type.

File: src/main/java/com/semmle/jira/addon/util/JiraUtils.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        return issueType;
+      }
+    }
+    return null;
+  }
+
+  public static void addWorkflowToProject(
+      String projectKey, String workflowName, String issueTypeName)
+      throws GenericEntityException, ProjectNotFoundException, IssueTypeNotFoundException {
+    WorkflowSchemeManager workflowSchemeManager = ComponentAccessor.getWorkflowSchemeManager();
```
Comment (aibaars) [inline src/main/java/com/semmle/jira/addon/util/JiraUtils.java]: I think this method should take a project ID as argument. That way you can move the `getProjectByCurrentKey` call out of this method. Same for issue type.

File: src/main/java/com/semmle/jira/addon/util/JiraUtils.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+    GenericValue workflowScheme = workflowSchemeManager.getWorkflowScheme(project);
+    IssueType lgtmIssueType = getIssueTypeByName(issueTypeName);
+    if (lgtmIssueType == null) {
+      throw new IssueTypeNotFoundException();
+    }
+    workflowSchemeManager.addWorkflowToScheme(workflowScheme, workflowName, lgtmIssueType.getId());
+  }
+
+  public static Status getLgtmWorkflowOpenStatus()
```
Comment (aibaars) [inline src/main/java/com/semmle/jira/addon/util/JiraUtils.java]: `getLgtmWorkflowOpenStatus` and `getLgtmWorkflowClosedStatus` are mostly duplicated methods. You could just rename the method to `getLgtmWorkflowStatus(String statusName)`

File: src/main/java/com/semmle/jira/addon/util/Constants.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package com.semmle.jira.addon.util;
+
+public class Constants {
+  public static final String workflowName = "LGTM alert";
```
Comment (aibaars) [inline src/main/java/com/semmle/jira/addon/util/Constants.java]: Please use `ALL_CAPS` style for constants.

File: src/main/resources/js/lgtm-addon.js [hunk 6]
```diff
@@ -304,6 +210,43 @@ function updateConfig() {
...
+				closeable : true,
+				fadeout : true
+			});
+			return;
+		} else if (jqXHR.getResponseHeader("Error") === "workflow") {
+			AJS.messages.error("#message-context", {
+				title : 'A problem happened when adding the LGTM alert workflow to the project.',
```
Comment (aibaars) [inline src/main/resources/js/lgtm-addon.js]: happened -> occurred

File: src/main/resources/js/lgtm-addon.js (hunk 1 above)
Comment (aibaars) [reply src/main/resources/js/lgtm-addon.js]: `doe` -> `does` 

Perhaps `The "LGTM alert" issue type could not be found.`

File: src/main/resources/js/lgtm-addon.js [hunk 7]
```diff
@@ -304,6 +210,43 @@ function updateConfig() {
 				fadeout : true
 			});
 			return;
+		} else if (jqXHR.getResponseHeader("Error") === "project") {
+			// This should not happen
+			AJS.messages.error("#message-context", {
+				title : 'The project ' + AJS.$(projectFieldId).select2('data').text
+						+ ' does not exist.',
```
Comment (aibaars) [inline src/main/resources/js/lgtm-addon.js]: Perhaps add `"` symbols around the project name.

**Pull request merged**